gsoc-2024-backend/src/main/java/backend/BackendVerticle.java

```diff
@@ -11,10 +17,46 @@ public static void main(String[] args) {
     vertx.deployVerticle(new BackendVerticle())
       .onFailure(Throwable::printStackTrace)
       .onSuccess(v -> System.out.println("Deployed backend"));
+
+  }
+  
+  public void getJoke(Promise<String > promise) {
```

Comment:
Since `client.send()` returns a `Future`, can your adapt the implementation to transform the future instead of using an intermediary promise?

Reply:
i will try tp remove intermediary promise.